source/qtmosquitto.cpp

```diff
@@ -103,6 +103,38 @@ bool QtMosquittoClient::doConnect(const QString& host, int port, int keepalive)
   return true;
 }
 
+bool QtMosquittoClient::doConnect(const QString &cafile, const QString &certfile, const QString &keyfile, const QString &host, int port, int keepalive)
+{
+    if (d->connected)
+    {
+      qWarning() << "QtMosquittoClient::doConnect: Already connected";
+      return false;
+    }
+
+    // Set TLS before connecting
+    QByteArray cafileBA(cafile.toUtf8());
+    QByteArray certfileBA(certfile.toUtf8());
+    QByteArray keyfileBA(keyfile.toUtf8());
+    int rt = mosquitto_tls_set(d->mosq, cafileBA.data(), NULL, certfileBA.data(), keyfileBA.data(), 0);
+    if (!(rt == MOSQ_ERR_SUCCESS))
+    {
+      qWarning() << "QtMosquittoClient::doConnect: Failed to set TLS" << rt;
+      return false;
+    }
+
+    // Connect
```

Comment:
This is duplicate code.

It would be better to just add a separate function "configureTls" that you call before the "doConnect".